cmd/wallet-sdk-js/test/openid4ci.js

```diff
@@ -0,0 +1,142 @@
+import '../src/wasm_exec.js';
+import Agent from '../src/index.js';
+
+
+describe("A suite is just a function", function () {
+
+    it("and so is a spec", async function () {
+        const env = __karma__.config;
+
+        const kmsDatabase = await CreateDB("test")
+        let agent = new Agent({
+            assetsPath: "base/dist",
+            kmsDatabase: kmsDatabase,
+            didResolverURI: "http://localhost:8072/1.0/identifiers",
+        });
+        await agent.initialize()
+
+        let didDoc = await agent.createDID({
+            didMethod: "ion",
+        })
+
+        console.log("INITIATE_ISSUANCE_URL: ", env.INITIATE_ISSUANCE_URL);
+
+        let openID4CIInteraction = await agent.createOpenID4CIIssuerInitiatedInteraction({
+            initiateIssuanceURI: env.INITIATE_ISSUANCE_URL
+        });
+
+        let userPINRequired = (await openID4CIInteraction.preAuthorizedCodeGrantParams()).userPINRequired;
+
+        expect(userPINRequired).toBe(false);
+
+        let credentials = await openID4CIInteraction.requestCredentialWithPreAuth({
+            pin: "",
+            didDoc: didDoc
+        });
+
+        expect(credentials.length).toBe(1);
+
+
+        let openID4VPInteraction = await agent.createOpenID4VPInteraction({
+            authorizationRequest: env.INITIATE_VERIFICATION_URL
+        })
+
+        let query = await openID4VPInteraction.getQuery();
+
+        let requirements = await agent.getSubmissionRequirements({
+            query: query,
+            credentials: credentials,
+        });
+
+        expect(requirements.length).toBe(1);
+        expect(requirements[0].descriptors.length).toBe(1);
+        expect(requirements[0].descriptors[0].matchedVCs.length).toBe(1);
+
+        await openID4VPInteraction.presentCredential({
+            credentials: requirements[0].descriptors[0].matchedVCs
+        })
```

Comment:
Might be cleaner to split the issuance and the presentation tests into 2 separate `it` clauses and save the `credentials` array globally in this file. Besides, any variables (`let`s mostly) that don't need to be changed after declaration can be changed to `const`s

Reply:
Const is fixed. 

Splitting into two sub-tests has some downsides. 
1. Changing the order of tests will break them.
2. It is not obvious to the reader of the code that one test depends on another test. 
3. We use the same single test approach for go, flutter, ios , and android test. This will be inconsistent